Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp

```diff
@@ -468,10 +541,26 @@ bool Connection::sendOutputMessage(UnixMessage& outputMessage)
     if (!attachments.isEmpty()) {
         int* fdPtr = 0;
 
+#if OS(ANDROID)
+        size_t attachmentFDBufferLength = 0;
+        for (const auto& attachment : attachments) {
+            switchOn(attachment,
+                [&attachmentFDBufferLength](const UnixFileDescriptor& fd) {
+                    if (!!fd)
+                        ++attachmentFDBufferLength;
+                },
+                [&hardwareBuffers](const RefPtr<AHardwareBuffer>& buffer) {
+                    if (buffer)
+                        hardwareBuffers.append(buffer);
+                }
+            );
+        }
+#else
```

Comment:
Something I don't like much of this function is that it will prepare the arrays of file descriptors, AHBs, and other minutiae every time the function is called. It would be interesting to know how often this is getting re-scheduled on `EAGAIN`/`EWOULDBLOCK` because it may be worth it to store all the “prepared” data (even including the `struct iov`, message header, and so on) as part of `Connection::m_pendingOutputMessage` and avoid re-calculating each time.

Anyway, this is just an idea that came to my mind while working on the Unix IPC code, for this PR it seems much better to keep the existing structure. Refactoring can be done later, if we think it's worth it.